File: SEP3_UserSystem/src/main/java/com/example/SEP3_UserSystem/model/domain/unit/user/UserStatus.java
package com.example.SEP3_UserSystem.model.domain.unit.user;

public class UserStatus {
    private boolean Online;
    private String Status;

    public UserStatus()
    {
        Online = false;
    }

    private UserStatus(boolean online, String status)
    {
        Online = online;
        Status = status;
    }

    public void login()
    {
        Online = true;
        Status = "online";
    }

    public void logoff()
    {
        Online = false;
    }

    public void setStatus(String status) {
        Status = status;
    }

    public String getStatus() {
        if (Online)
        {
            return "[" + Status + "]";
        }
        else
        {
            return "[offline]";
        }
    }

    public boolean isOnline() {
        return Online;
    }

    public UserStatus copy()
    {
        return new UserStatus(Online,Status);
    }
}

File: SEP3_Client1/Model/Unit/User/UserStatus.cs
namespace SEP3_Client.Model.Unit.User
{
    public class UserStatus
    {
        public bool Online { get; set; }
        public string Status { get; set; }
        
        public UserStatus()
        {
            Online = false;
        }

        public void Login()
        {
            Online = true;
            Status = "online";
        }

        public void Logoff()
        {
            Online = false;
        }

        public void SetStatus(string status) {
            Status = status;
        }

        public string GetStatus() {
            if (Online)
            {
                return "[" + Status + "]";
            }
            else
            {
                return "[offline]";
            }
        }

        public bool IsOnline() {
            return Online;
        }

        public UserStatus Copy()
        {
            return new UserStatus
            {
                Online = Online,
                Status = Status
            };
        }
    }
}